### PR TITLE
Feature/spilo fixes

### PR DESCRIPTION
**Pull request by feikesteenbergen**

### CODE REVIEW

**Comment by hjacobs** [comment]
@feikesteenbergen could you tackle the decreasing code coverage please?

File: senza/templates/postgresapp.py [hunk 1]
```diff
@@ -190,10 +203,10 @@
           Version: "2012-10-17"
           Statement:
           - Effect: Allow
-            Action: s3:*
+            Action:
+              - s3:*
             Resource:
-              - arn:aws:s3:::{{wal_s3_bucket}}/spilo/*
-              - arn:aws:s3:::{{wal_s3_bucket}}
+              - "arn:aws:s3:::zalando-acid-eu-west-1-spilo-app/spilo/{{Arguments.version}}/*"
```
Comment (alexeyklyukin) [inline senza/templates/postgresapp.py]: Should be {{wal_s3_bucket}}

File: senza/templates/postgresapp.py [hunk 2]
```diff
@@ -208,6 +221,81 @@
...
+  {{#add_replica_loadbalancer}}
+  SpiloReplicaSG:
+    Type: "AWS::EC2::SecurityGroup"
+    Properties:
+      GroupDescription: "Security Group for the replica ELB"
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: {{postgres_port}}
+          ToPort: {{postgres_port}}
+          CidrIp: 0.0.0.0/0
```
Comment (alexeyklyukin) [inline senza/templates/postgresapp.py]: Should we do the same for the replica ELB as we do for the master one? I think securing the database vs unauthorized reads is as important as securing it vs unauthorized writes

File: senza/templates/postgresapp.py [hunk 3]
```diff
@@ -208,6 +221,81 @@
...
+          SourceSecurityGroupId:
+            Fn::GetAtt:
+              - SpiloReplicaSG
+              - GroupId
+        {{/add_replica_loadbalancer}}
+
+        - IpProtocol: tcp
+          FromPort: 0
+          ToPort: 65535
+          SourceSecurityGroupName: "Odd (SSH Bastion Host)"
```
Comment (alexeyklyukin) [inline senza/templates/postgresapp.py]: Is the name really hard-coded, or is there a function that gets it?

File: senza/templates/postgresapp.py [hunk 4]
```diff
@@ -208,6 +221,81 @@
...
+    Type: "AWS::EC2::SecurityGroup"
+    Properties:
+      GroupDescription: "Security Group for Spilo members"
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: {{postgres_port}}
+          ToPort: {{postgres_port}}
+          SourceSecurityGroupId:
+            Fn::GetAtt:
+              - SpiloMasterSG
```
Comment (alexeyklyukin) [inline senza/templates/postgresapp.py]: What if the master group {{elb_access_cidr}} does not include the address range of spilo members, wouldn't it break the replication?

File: senza/templates/postgresapp.py (hunk 3 above)
Comment (feikesteenbergen) [inline senza/templates/postgresapp.py]: It is hardcoded, will convert it into a function during senza init, together with the Security Group for the zmon-agent.

File: senza/templates/postgresapp.py (hunk 4 above)
Comment (feikesteenbergen) [inline senza/templates/postgresapp.py]: That would be no problem, as the SpiloMemberIngressMembers resource defines that all members of the SpiloMemberSG can communicate freely over all the ports

**Comment by feikesteenbergen** [comment]
Closing (temporarily)

**Comment by alexeyklyukin** [comment]
+1

**Comment by hjacobs** [comment]
:+1:

**Pull request merged**